inspirehep/modules/records/views.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from flask import Blueprint, jsonify
+
+from invenio_base.decorators import wash_arguments
+from invenio_ext.es import es
+from invenio_search.api import Query
+
+
+blueprint = Blueprint(
+    'inspire_records',
+    __name__,
+    url_prefix='/record',
+    template_folder='templates',
+    static_folder="static",
+)
+
+
+@blueprint.route('/api/impactgraph', methods=['GET'])
+@wash_arguments({'recid': (unicode, "")})
+def impactgraph(recid):
+    """Handler for impact graph information retrieval."""
+    record = es.get_source(index='hep', id=recid, doc_type='record')
+
+    out = {}
+
+    # Add information about current record
+    out['recid'] = record['control_number']
+    out['title'] = record['breadcrumb_title']
+    out['year'] = record['earliest_date'].split('-')[0]
+
+    # Get citations
+    citations = []
+
+    es_query = Query('refersto:' + recid).search()
+    es_query.body.update(
+        {
+            'size': 9999
+        }
+    )
```

Comment:
But Invenio-search doesn't know about references... Do you mean just the possibility to override the size?